src/value.rs

```diff
@@ -27,35 +29,152 @@ impl NumericValue {
     }
 }
 
-#[derive(PartialEq, Debug)]
+impl PartialOrd for NumericValue {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        match (self.0.as_i64(), self.0.as_u64(), self.0.as_f64()) {
+            (Some(a), _, _) => match (other.0.as_i64(), other.0.as_u64(), other.0.as_f64()) {
+                (Some(b), _, _) => Some(a.cmp(&b)),
+                (_, Some(b), _) => {
+                    if a < 0 {
+                        Some(std::cmp::Ordering::Less)
+                    } else {
+                        (a as u64).partial_cmp(&b)
+                    }
+                }
+                (_, _, Some(b)) => (a as f64).partial_cmp(&b),
+                _ => None,
+            },
+            (_, Some(a), _) => match (other.0.as_i64(), other.0.as_u64(), other.0.as_f64()) {
+                (_, Some(b), _) => Some(a.cmp(&b)),
+                (Some(b), _, _) => {
+                    if b < 0 {
+                        Some(std::cmp::Ordering::Greater)
+                    } else {
+                        a.partial_cmp(&(b as u64))
+                    }
+                }
+                (_, _, Some(b)) => (a as f64).partial_cmp(&b),
+                _ => None,
+            },
+            (_, _, Some(a)) => match (other.0.as_i64(), other.0.as_u64(), other.0.as_f64()) {
+                (_, _, Some(b)) => a.partial_cmp(&b),
+                (Some(b), _, _) => a.partial_cmp(&(b as f64)),
+                (_, Some(b), _) => a.partial_cmp(&(b as f64)),
+                _ => None,
+            },
+            _ => None, // not representable using rust types
+        }
+    }
+}
+
+#[derive(PartialEq, Debug, Clone)]
 pub enum Value {
     Numeric(NumericValue),
     String(String),
     Boolean(bool),
 }
 
+impl FromStr for NumericValue {
+    type Err = serde_json::Error;
+
+    fn from_str(value: &str) -> Result<Self, Self::Err> {
+        Ok(NumericValue(value.parse()?))
+    }
+}
```

Comment:
As we use this internally too, I needed to return `serde_json::Error` instead of `crate::Error` to avoid circular dependency in segment evaluation.
I provide
```
    #[error(transparent)]
    SerdeError(#[from] serde_json::Error),
```
to hopefully cope with that